contracts/bitizens/AvatarItemOperator.sol

```diff
@@ -1,48 +1,132 @@
+    emit BurnToken(msg.sender, _tokenId);
+  }
+
+  function createToken( 
     address _owner,
-    uint32 _listNumber,
-    uint8 _setNumber,
-    bool _isBitizenItem, 
-    bool _isSet, 
-    uint8 _types,
-    uint8 _rarity,
-    uint8 _socket,
-    uint8 _gender,
-    uint8 _ext1,
-    uint8 _ext2
-    ) external onlyOperator returns(uint256 _tokenId)  {
-    _tokenId = itemService.createAvatarItem(_owner,_listNumber,_setNumber,_isBitizenItem,_isSet,_types,_rarity,_socket,_gender,_ext1,_ext2);
-    emit AvatarItemCreateSuccess(msg.sender, _tokenId);
-  }
-
-  function getOwnedTokenIds() external view returns(uint256[] _tokenIds){
-    _tokenIds = itemService.getOwnedTokenIds(msg.sender);
-  }
-
-  function getAvatarItemInfo(uint256 _tokenId)
+    address _founder,
+    address _creator,
+    bool _isBitizenItem,
+    int16 _probability,
+    uint256[4] _attr1,
+    uint8[5] _attr2) 
     external
-    view 
-    returns(
-    uint32 _listNumber,
-    uint16 _setNumber,
-    bool _isBitizenItem, 
-    bool _isSet,
-    uint8[6] _attr){
-    (_listNumber,_setNumber,_isBitizenItem,_isSet,_attr) = itemService.getAvatarItemInfo(_tokenId);
+    onlyOwner
+    returns(uint256 _tokenId) {
+    _tokenId = _mintToken(_owner, _founder, _creator, _isBitizenItem, _probability, _attr1, _attr2);
+    emit CreateToken(_owner, _tokenId);
+  }
+  
+  function batchCreateToken(
+    address _owner,
+    bool[] _isBitizenItems,  // base length
+    int16[] _probabilities,  // one time
+    address[] _addresses,    // two times 
+    uint256[] _attrs1,       // four times 
+    uint8[] _attrs2          // five times
+    )
+    external 
+    onlyOwner
+    returns(uint256[] _tokenIds) {
+    // enure the array len are valid
+    require(_isBitizenItems.length > 0);
+    require(_isBitizenItems.length * 1 == _probabilities.length);
+    require(_isBitizenItems.length * 2 == _addresses.length);
+    require(_isBitizenItems.length * 4 == _attrs1.length);
+    require(_isBitizenItems.length * 5 == _attrs2.length);
+    _tokenIds = new uint256[](_isBitizenItems.length);
+    for(uint8 i = 0; i < _isBitizenItems.length; i++) {
+      _tokenIds[i] = _mintToken(
+        _owner,
+        _addresses[i * 2 + 0],
+        _addresses[i * 2 + 1],
+        _isBitizenItems[i],
+        _probabilities[i],
+        _convertToAttr1(i, _attrs1), 
+        _convertToAttr2(i, _attrs2)
+      );
+    }
+    emit BatchCreateToken(_owner, _tokenIds);
+  }
+
+  function _mintToken( 
+    address _owner,
+    address _founder,
+    address _creator,
+    bool _isBitizenItem,
+    int16 _probability,
+    uint256[4] _attr1,
+    uint8[5] _attr2) 
+    private 
+    returns(uint256 _tokenId) {
+    _tokenId = itemService.createToken(_owner, _founder, _creator, _isBitizenItem, _probability, _attr1, _attr2);
+  }
+
+  function _convertToAttr1(uint8 _index, uint256[] _attrs1) private pure returns (uint256[4] _attr2) {
+    _attr2[0] = _attrs1[_index * 4 + 0];
```

Comment:
Same as above, while I understand it looks formatted, but there's no need to do the "+ 0" here.